workflow/rocoto_funcs/base.py

```diff
@@ -268,13 +267,13 @@ def xml_task(xmlFile,expdir,task_id,cycledefs,dcTaskEnv={},dependencies="",metat
     'command': f'&HOMErrfs;/workflow/sideload/launch.sh JRRFS_'+f'{command_id}'.upper(),
     'join': f'&LOGROOT;/rrfs{WGF}.@Y@m@d/@H/{task_id}_{TAG}_@Y@m@d@H.log',
     'jobname': f'{TAG}_{task_id}_c@H',
-    'account': get_cascade_env(f'ACCOUNT_{task_id}'.upper()),
-    'queue': get_cascade_env(f'QUEUE_{task_id}'.upper()),
-    'partition': get_cascade_env(f"PARTITION_{task_id}".upper()),
-    'walltime': get_cascade_env(f"WALLTIME_{task_id}".upper()),
-    'nodes': get_cascade_env(f"NODES_{task_id}".upper()),
-    'reservation': get_cascade_env(f"RESERVATION_{task_id}".upper()),
-    'native': get_cascade_env(f"NATIVE_{task_id}".upper())
+    'account': get_cascade_env(f'ACCOUNT_{meta_id}'.upper()),
+    'queue': get_cascade_env(f'QUEUE_{meta_id}'.upper()),
+    'partition': get_cascade_env(f"PARTITION_{meta_id}".upper()),
+    'walltime': get_cascade_env(f"WALLTIME_{meta_id}".upper()),
+    'nodes': get_cascade_env(f"NODES_{meta_id}".upper()),
+    'reservation': get_cascade_env(f"RESERVATION_{meta_id}".upper()),
+    'native': get_cascade_env(f"NATIVE_{meta_id}".upper())
```

Comment:
I think we don't need to make changes in lines 270-276.  Have you run into any problems with the original `task_id`?

Reply:
It was using "task_id" but for tasks with longer strings such as ic_m001, it can't find the right resources. On the other hand, we can use "meta_id" which is exactly the task name, such as "ic" and "lbc".

Reply:
@chunhuazhou Whether you specify `NODES_IC_M001` or not, the `get_cascade_env` function will make sure we will get the correct environmental setting in a reverse cascade way, i.e. if `NODES_IC_M001` is not available, the function will then check `NODES_IC`. The function is designed to have the maximum granularity for potential fine tuning. We don't have to use it at this stage, but let's keep this capability.